### DISCUSSION TITLE
Uniquely defining an Ensemble Member

### DISCUSSION THREAD

**Discussion by kobebryant432** [CORDEX-be2/ValEnsPy#14] [Q&A]
What uniquely defines an ensemble member? We need to have an exhaustive list. These should be explicit or implicit attributes of an ensemble member object.

Currently, the following characteristics are considered:
Explicitly (i.e. as an attribute of the object):
- experiment (flexible as to how this is defined, e.g. ssp5.8.5 or veg_option_DIFF)
- institution (which institute produced this model data)
- model name
- run_id (The unique identifier of a run, e.g. r1i1p1f1)

Implicitly (i.e. as characteristics that are commonly present in the x-array datasets that make up the ensemble member)
- time_period considered (As the unique period for which all variables are defined)
- domain 
- resolution 

> [!IMPORTANT]  
> Are we missing any or is this too constrictive?

@vergauwenthomas maybe for an NWP case there are other characteristics (i.e. lead time?).

**Comment by vergauwenthomas** [comment]
For NWP,  there are indeed two timestamps (dims) to uniqueify an atmospheric state:

- the validate (same as for climate applications)
- the base data (this varies for nwp in contrast to climate applications)

I suggest to not take this in consideration for now, and add this feature later. If we could think of a way to add new `explicit` and `implicit` dimensions in general, we can easily adapt it to match the application IMO.

**Comment by Ivanderkelen** [comment]
What you list is pretty complete Kobe. As a first instance, I would focus on climate simulations. 
More inspiration on metadata, and attributes to define an ensemble member can be found in the CORDEX protocol: https://is-enes-data.github.io/cordex_archive_specifications.pdf
